pages/game/flash-card.tsx

```diff
@@ -1,60 +1,113 @@
-//User can play learning page.
+import React, { useState, useEffect, createContext } from 'react';
+import { phraseType, learningContextType } from '../../types/types';
+import FlashCard from '../../components/game/flashCard/FlashCard';
+import { getSession } from 'next-auth/react';
 
-import React, { useState, useEffect, createContext } from "react";
-import { phraseType, learningContextType } from "../../types/types";
-import PhraseList from "../../components/game/flashCard/PhraseList";
-import { FlashCard } from "../../components/game/flashCard/FlashCard";
+export const SelectedMaterial = createContext<LearningContextProps | null>(null);
 
-const BACKEND_URL = process.env.NEXT_PUBLIC_BACKEND_URL;
-const TOPICS = ["First Contact", "Playing together", "Drinking"]
-export const SelectedMaterial = createContext<learningContextType | null>(null)
+interface LearningContextProps {
+  topic: string;
+  phrases: phraseType[];
+  correctPhrases: number[];
+  storeCorrectPhrases: (id: number) => void;
+  resetGame: () => void;
+}
+
+
+interface ContentType {
+  content_id: number;
+  japanese_slang: string;
+  english_slang: string;
+  formal_version: string;
+  description: string;
+}
+
+
+//shuffle functions to ramdom the cards
+const shuffleArray = (array: phraseType[]): phraseType[] => {
+  let shuffledArray = [...array];
+  for (let i = shuffledArray.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [shuffledArray[i], shuffledArray[j]] = [shuffledArray[j], shuffledArray[i]];
+  }
+  return shuffledArray;
+};
 
 const FlashCardPage = () => {
-  const [selectedTopic, setSelectedTopic] = useState(TOPICS[0])
-  const [selectedPhrase, setSelectedPhrase] = useState<phraseType>({
-    id: 0,
-    japanese: "loading...",
-  })
+  const [phrases, setPhrases] = useState<phraseType[]>([]);
+  const [correctPhrases, setCorrectPhrases] = useState<number[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [userInfo, setUserInfo] = useState<any>("");
+
+  //get user info
+  // useEffect(() => {
+  //   getSession().then((session) => {
+  //     const newGameState: any = {
+  //       message: "Test",
+  //       FlashCardPage: null,
+  //       playerName: session?.user.username || "Player",
+  //       playerId: session?.user.pk || 0,
+  //   };
+  //     setUserInfo(newGameState.playerId);
+  //  });
+  // }, []);
+
+  // Fetch cards
+  const fetchPhrases = async () => {
+    setLoading(true); // Set loading to true
+    try {
+      const response = await fetch(`http://localhost:8000/api/flash-card?limits=10&user-id=4`);
```

Comment:
use insted of the direct path this getFetchBackendURL('/api/flash-card?limits=10&user-id=4')